ast/src/main/java/com/graphicsfuzz/common/ast/type/QualifiedType.java

```diff
@@ -79,6 +80,34 @@ public void replaceQualifier(TypeQualifier oldQualifier, TypeQualifier newQualif
     qualifiers.add(newQualifier);
   }
 
+  /**
+   * Sets or replaces location qualifier.
+   *
+   * @param location Location to set in the location qualifier.
+   */
+  public void setLocationQualifier(int location) {
+    Optional existingLayoutQualifierSequence = Optional.empty();
+    List<LayoutQualifier> qualifierList = new ArrayList<LayoutQualifier>();
```

Comment:
Nit: change ArrayList<LayoutQualifier> to ArrayList<>.  I don't think this is required by Google Java style but it's a style I like to use for this project (also IDEA recommends it).

Reply:
Done